wfe2/wfe.go

```diff
@@ -2337,7 +2337,7 @@ func (wfe *WebFrontEndImpl) NewOrder(
 		names[i] = ident.Value
 	}
 
-	err = policy.WellFormedDomainNames(names)
+	err = policy.WellFormedDomainNames(core.UniqueLowerNames(names))
```

Comment:
My only question is: should this be
```suggestion
	names = core.UniqueLowerNames(names)
	err = policy.WellFormedDomainNames(names)
```

Is there a strong reason to carry around the non-unique-lower-names through the rest of the WFE code, only to re-unique-and-lower them again in the RA?

To be clear, I'm totally fine with doing the minimal fix here and now, but I think we should consider this.

Reply:
This change would require modifying the contract between WFE.NewOrder and RA.NewOrder to expect normalized names. Such an adjustment would be part of a two-phase update, where RA will eventually stop normalizing names, avoiding redundancy. This isn't necessarily a bad idea, I actually quite like it, but it might not be suitable for inclusion in the current PR.

Reply:
Filed: https://github.com/letsencrypt/boulder/issues/7555

Reply:
I changed `wfe.NewOrder` to use its copy of the normalized names while we deliberate about the contract.